### DISCUSSION TITLE
Lesson 3: Question about Interface

### DISCUSSION THREAD

**Discussion by programonauta** [smartcontractkit/full-blockchain-solidity-course-py#539] [Q&A] [answered]
Hi, I have two questions about Interfaces

1) As I could understood, Interface works like a declaration as used in Header files, for instance in C Language. But where  are exactly these functions like `version()` or `latestRoundData()` defined? i.e. where can I find their source codes?

2) In the line `@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol"` how and where Solidity find the file `AggregatorV3Inteface.sol`? I mean, if I'd like to explore the interface and all the functions it has, how to find it?

Thanks

**Comment by cromewar** [comment, marked as answer]
Hello @programonauta 

1. they are defined on their own contract file, the interfaces are abstract contracts which should not be deployed just used by other contracts.
2. Remix IDE automatically recognize the `@` identifier, they actually refer to Github repositories, if you want to look in deep about that you can check [chainlink repo](https://github.com/smartcontractkit/chainlink), here you'll find all you need.

**Reply by programonauta** [reply]
Thanks @cromewar 

About your answer of Question 1, I'm not sure if I got it. So, for instance, the source code of version() function will be in any Solidity (`.sol`) file on this directory: https://github.com/smartcontractkit/chainlink/tree/develop/contracts/src/v0.6?

**Reply by cromewar** [reply]
actually is on the exact same path you are importing from  `contracts/src/v0.6/interfaces/AggregatorV3Interface.sol`
As you can see bellow.
![image](https://user-images.githubusercontent.com/20199180/145694472-67db8dad-1ee1-4cc1-a2be-ab8a6adb1c3c.png)
So, yes all the interfaces must have a `.sol` file stored somewhere, using the imports you are just taking them from github repos.

**Reply by programonauta** [reply]
Yes, I got it, but the image shows only the declaration, not the source code it self. Is the source code available somewhere? 

If not, where is the "binary"? and how the compiler find out it?

Sorry if there is some prereq concept I'm not understanding :)

**Reply by cromewar** [reply]
when you say "the declaration" you actually refer to the source code... So yep this is ended the source code.
The binary files compiles to something called the ABI (Application Binary Interface) which is a low level stuff and Remix is smart enough to handle it for you, as you are going to see later on the course this is managed with a library called Web3 (on javascript or python) and you can use it directly or with a framework like brownie.
